test/functional/test_framework/mininode.py

```diff
@@ -9,1462 +9,252 @@
 This python code was modified from ArtForz' public domain  half-a-node, as
 found in the mini-node branch of http://github.com/jgarzik/pynode.
 
-NodeConn: an object which manages p2p connectivity to a bitcoin node
-NodeConnCB: a base class that describes the interface for receiving
-            callbacks with network messages from a NodeConn
-CBlock, CTransaction, CBlockHeader, CTxIn, CTxOut, etc....:
-    data structures that should map to corresponding structures in
-    bitcoin/primitives
-msg_block, msg_tx, msg_headers, etc.:
-    data structures that represent network messages
-ser_*, deser_*: functions that handle serialization/deserialization
-"""
-
+P2PConnection: A low-level connection object to a node's P2P interface
+P2PInterface: A high-level interface object for communicating to a node over P2P"""
 import asyncore
-from codecs import encode
 from collections import defaultdict
-import copy
-import hashlib
 from io import BytesIO
 import logging
-import random
 import socket
 import struct
 import sys
-import time
 from threading import RLock, Thread
 
-from test_framework.siphash import siphash256
-from test_framework.util import hex_str_to_bytes, bytes_to_hex_str, wait_until
-
-BIP0031_VERSION = 60000
-MY_VERSION = 70014  # past bip-31 for ping/pong
-MY_SUBVERSION = b"/python-mininode-tester:0.0.3/"
-MY_RELAY = 1 # from version 70001 onwards, fRelay should be appended to version messages (BIP37)
-
-MAX_INV_SZ = 50000
-MAX_BLOCK_BASE_SIZE = 1000000
+from test_framework.primitives import *
```

Comment:
Did we want to avoid `*` imports or is this a case that merits exception?

Reply:
Yeah, it's ugly. See the commit message for 64d947b82df924c101f0fd00f4891177a2d0ed44:

```
mininode.py wildcard imports all names from primitives.py. This is
to avoid having to change all test scripts that import from mininode.py.
```

I don't like wildcard imports, but we're pretty lax about them in the functional tests, so this isn't making things much worse.

(I'd be happy to remove wildcard imports entirely and have tried to do that before in #9876 and #9943)

Reply:
Ah I see, thanks for the pointer. I'd be happy to help with this after the more important `P2P*` changes are merged (unless there's some good reason to do the two simultaneously).